### ISSUE TITLE
Compile error on Alpine Linux w/ZFS 0.7.0-RC5 

### ISSUE THREAD

**Issue by matjohn2** [zfsonlinux/zfs#6357]
### System information
Type                                | Version/Name
  ---                                  |     --- 
Distribution Name       | Alpine
Distribution Version    | 3.5.2
Linux Kernel                 | Linux fbba654516a3 4.9.31-moby
Architecture                 | x64
ZFS Version                  | 0.7.0-RC5
SPL Version                  | 0.7.0-RC5

### Describe the problem you're observing
Cannot compile 0.7.0-RC5 on Alpine 
(using 0.7.0-RC5 to work around this issue #2567 compiling 0.6.5.11)

Error output for `make` is below.

### Describe how to reproduce the problem

```
cd /zfs
./autogen.sh
./configure
make
```

### Include any warning/errors/backtraces from the system logs

```
  CC       fm.lo
  CC       gzip.lo
  CC       lzjb.lo
  CC       lz4.lo
  CC       metaslab.lo
  CC       mmp.lo
In file included from ../../include/sys/refcount.h:31:0,
                 from ../../include/sys/abd.h:32,
                 from ../../module/zfs/mmp.c:25:
../../module/zfs/mmp.c: In function 'mmp_thread':
../../module/zfs/mmp.c:449:47: error: 'HZ' undeclared (first use in this function)
       ((next_time - gethrtime()) / (NANOSEC / HZ)));
                                               ^
../../include/sys/zfs_context.h:367:60: note: in definition of macro 'cv_timedwait_sig'
 #define cv_timedwait_sig(cv, mp, at)  cv_timedwait(cv, mp, at)
                                                            ^~
../../module/zfs/mmp.c:449:47: note: each undeclared identifier is reported only once for each function it appears in
       ((next_time - gethrtime()) / (NANOSEC / HZ)));
                                               ^
../../include/sys/zfs_context.h:367:60: note: in definition of macro 'cv_timedwait_sig'
 #define cv_timedwait_sig(cv, mp, at)  cv_timedwait(cv, mp, at)
                                                            ^~
make[3]: Leaving directory '/zfs/lib/libzpool'
make[3]: *** [Makefile:906: mmp.lo] Error 1
make[2]: *** [Makefile:548: all-recursive] Error 1
make[2]: Leaving directory '/zfs/lib'
make[1]: *** [Makefile:720: all-recursive] Error 1
make[1]: Leaving directory '/zfs'
make: *** [Makefile:589: all] Error 2
```

**Comment by matjohn2** [comment]
Alpine's libc doesn't expose `HZ`, problem can be worked around using the following in `mmp.c`;

```
#ifndef HZ
#define HZ 1000
#endif
```

However, before I submit a PR, I'd like to discuss the implications of manually specifying a value for `HZ`, it seems to be used in a lot of calculations within`mmp.c`; could this have negative impact on performance, stability or data consistency?

Thanks.

**Comment by ofaaland** [comment]
@matjohn2,

Thanks for reporting this.   This mmp.c code is very new, and this use of `HZ` was an error - the rest of the code uses `hz` as that is what Illumos provides, and the SPL defines `hz=HZ` to accomodate that.  The mmp.c code should have used `hz` to be consistent.

For userspace builds (affecting tools ztest, zdb, and zhack), it does end up getting set to a constant value.  As long as mmp uses `hz` instead of `HZ` so it is consistent, the wait times etc. should all be correct as the code is all using the same ratio of ticks to nanoseconds.

Would you try this?
`
--- a/module/zfs/mmp.c
+++ b/module/zfs/mmp.c
@@ -446,7 +446,7 @@ mmp_thread(spa_t *spa)
                CALLB_CPR_SAFE_BEGIN(&cpr);
                (void) cv_timedwait_sig(&mmp->mmp_thread_cv,
                    &mmp->mmp_thread_lock, ddi_get_lbolt() +
-                   ((next_time - gethrtime()) / (NANOSEC / HZ)));
+                   ((next_time - gethrtime()) / (NANOSEC / hz)));
                CALLB_CPR_SAFE_END(&cpr, &mmp->mmp_thread_lock);
        }
`

**Comment by matjohn2** [comment]
Thanks @ofaaland, I'll try that now and report back. 

My hack of defining HZ manually got me past the issue, but i'm now having another alpine libc vs glibc issue as with `libintl_gettext` when compiling `zfs-tests`. 

I've tried adding `LDFLAGS=-liconv -lintl` without success.

(if you want another issue raising, no problems, but I thought you may have some insights).

```
make[5]: Entering directory '/zfs/tests/zfs-tests/cmd/mkbusy'
  CC       mkbusy.o
  CCLD     mkbusy
make[5]: Leaving directory '/zfs/tests/zfs-tests/cmd/mkbusy'
Making all in mkfile
make[5]: Entering directory '/zfs/tests/zfs-tests/cmd/mkfile'
  CC       mkfile.o
  CCLD     mkfile
mkfile.o: In function `usage':
/zfs/tests/zfs-tests/cmd/mkfile/mkfile.c:271: undefined reference to `libintl_gettext'
mkfile.o: In function `main':
/zfs/tests/zfs-tests/cmd/mkfile/mkfile.c:110: undefined reference to `libintl_gettext'
/zfs/tests/zfs-tests/cmd/mkfile/mkfile.c:184: undefined reference to `libintl_gettext'
/zfs/tests/zfs-tests/cmd/mkfile/mkfile.c:132: undefined reference to `libintl_gettext'
/zfs/tests/zfs-tests/cmd/mkfile/mkfile.c:138: undefined reference to `libintl_gettext'
mkfile.o:/zfs/tests/zfs-tests/cmd/mkfile/mkfile.c:220: more undefined references to `libintl_gettext' follow
collect2: error: ld returned 1 exit status
make[5]: *** [Makefile:585: mkfile] Error 1
make[5]: Leaving directory '/zfs/tests/zfs-tests/cmd/mkfile'
make[4]: *** [Makefile:562: all-recursive] Error 1
make[4]: Leaving directory '/zfs/tests/zfs-tests/cmd'
make[3]: *** [Makefile:541: all-recursive] Error 1
make[3]: Leaving directory '/zfs/tests/zfs-tests'
make[2]: *** [Makefile:541: all-recursive] Error 1
make[2]: Leaving directory '/zfs/tests'
make[1]: *** [Makefile:720: all-recursive] Error 1
make[1]: Leaving directory '/zfs'
make: *** [Makefile:589: all] Error 2
```

**Comment by matjohn2** [comment]
Compilation succeeded past `mmp.c` with your proposed change @ofaaland 👍 

Now just stuck at the `libintl_gettext` issue above.

**Comment by ofaaland** [comment]
@matjohn2,
Thanks for testing that.

I'll see what I can find out about the libintl_gettext issue you encountered.  Please do create a separate issue for it, if you don't mind, and delete that comment from this issue as I've referred to this issue in the PR for the HZ->hz change.

Thanks!
